mindmeld/text_preparation/text_preparation_pipeline.py

```diff
@@ -627,3 +662,21 @@ def _construct_pipeline_component(  # pylint: disable=W0640
             raise TypeError(
                 f"{component} must be of type String or {expected_component_class.__name__}."
             )
+
+
+# Borrowed from https://chadrick-kwag.net/json-dumpingserializing-custom-python-classes/
+class TPPJSONEncoder(json.JSONEncoder):
+    """
+    Custom Encoder class defined to obtain recursive JSON representation of a TextPreparationPipeline.
+
+    Args:
+        None.
+
+    Returns:
+        Custom JSON Encoder class (json.JSONEncoder) .
+    """
+
+    def default(self, o):
+        if "tojson" in dir(o):
```

Comment:
We can can make this a bit more robust by check specific conditions we need to satisfy.
```python
tojson = getattr(o, 'tojson', None)
if callable(tojson):
    return tojson()
```

Reply:
Nice! and done!